processing/src/main/java/org/apache/druid/query/lookup/LookupSegment.java

```diff
@@ -51,11 +51,11 @@ public LookupSegment(final String lookupName, final LookupExtractorFactory looku
         Sequences.simple(() -> {
           final LookupExtractor extractor = lookupExtractorFactory.get();
 
-          if (!extractor.canIterate()) {
-            throw new ISE("Cannot iterate lookup[%s]", lookupExtractorFactory);
+          if (!extractor.supportsAsMap()) {
+            throw new ISE("Cannot retrieve map view from lookup[%s]", lookupExtractorFactory);
           }
 
-          return extractor.iterable().iterator();
+          return extractor.asMap().entrySet().iterator();
```

Comment:
well, i guess if we wanted to support other things maybe in the future we should add back an iterator later that spits out something other than map entry (result row or something perhaps?) to use for the direct queries

Reply:
In theory, I suppose, but there aren't any examples in the code base where it would be useful. I thought about keeping the flexibility in case some extensions found it useful, but decided against it, because I'm not aware of any and I didn't want to keep complexity for unclear benefit.